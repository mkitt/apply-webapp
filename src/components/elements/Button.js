// @flow
import glamorous from 'glamorous'
import { defaultProps } from 'recompose'
import { COLORS } from '../../constants/theme'

const box = {
  minWidth: 240,
  height: 48,
  paddingLeft: 32,
  paddingRight: 32,
  fontSize: 14,
  textTransform: 'uppercase',
}

const primary = {
  color: COLORS.WHITE,
  backgroundColor: COLORS.GREY_4,
}

const secondary = {
  color: COLORS.WHITE,
  backgroundColor: COLORS.GREY_9,
}

export default defaultProps({
  role: 'button',
})(glamorous.button(
  {
    textAlign: 'left',
  },
  props => ({
    marginTop: props.mt || props.marginTop,
    color: props.color,
    backgroundColor: props.backgroundColor,
    ...(props.box ? box : {}),
    ...(props.primary ? primary : {}),
    ...(props.secondary ? secondary : {}),
  }),
))
