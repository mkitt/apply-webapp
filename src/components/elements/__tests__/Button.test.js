import React from 'react'
import renderer from 'react-test-renderer'
import Button from '../Button'

describe('<Button />', () => {
  it('shows the correct snapshot tree for the <Button> element', () => {
    const component = <Button>The Button</Button>
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot tree for the <Button> element with composable options', () => {
    const component = (
      <Button box centered color="pink" mt={8}>
        The Composable Button
      </Button>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot tree for the <Button> element with primary option', () => {
    const component = <Button primary >The Primary Button</Button>
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot tree for the <Button> element with secondary option', () => {
    const component = <Button secondary >The Secondary Button</Button>
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
