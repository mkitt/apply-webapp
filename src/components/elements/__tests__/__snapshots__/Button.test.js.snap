// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> shows the correct snapshot tree for the <Button> element 1`] = `
.glamor-0,
[data-glamor-0] {
  text-align: left;
}

<button
  className="glamor-0"
  role="button"
>
  The Button
</button>
`;

exports[`<Button /> shows the correct snapshot tree for the <Button> element with composable options 1`] = `
.glamor-0,
[data-glamor-0] {
  text-align: center;
  min-width: 240px;
  height: 48px;
  padding-left: 32px;
  padding-right: 32px;
  font-size: 14px;
  text-transform: uppercase;
  margin-top: 8px;
  color: pink;
}

<button
  className="glamor-0"
  role="button"
>
  The Composable Button
</button>
`;

exports[`<Button /> shows the correct snapshot tree for the <Button> element with primary option 1`] = `
.glamor-0,
[data-glamor-0] {
  text-align: left;
  color: #fff;
  background-color: #4d4d4d;
}

<button
  className="glamor-0"
  role="button"
>
  The Primary Button
</button>
`;

exports[`<Button /> shows the correct snapshot tree for the <Button> element with secondary option 1`] = `
.glamor-0,
[data-glamor-0] {
  text-align: left;
  color: #fff;
  background-color: #999;
}

<button
  className="glamor-0"
  role="button"
>
  The Secondary Button
</button>
`;
